app/workers/fast_track/disability_compensation_job.rb

```diff
@@ -52,16 +51,47 @@ def perform(form526_submission_id, full_name)
       end
     end
 
+    # class AccountNotFoundError < RuntimeError; end
+
     private
 
-    def bp_readings(client)
-      @bp_readings ||= client.list_resource('observations')
-      @bp_readings.present? ? FastTrack::HypertensionObservationData.new(@bp_readings).transform : []
-    end
+    class LighthouseVetsHealth
+      def initialize(form526_submission)
+        @form526_submission = form526_submission
+      end
+
+      def client
+        @client ||= Lighthouse::VeteransHealth::Client.new(account.icn.presence)
+      end
+
+      def account
+        user_uuid = @form526_submission.user_uuid.presence
+        edipi = @form526_submission.auth_headers['va_eauth_dodedipnid'].presence
+        account = Account.find_by(idme_uuid: user_uuid) if user_uuid
+        account ||= Account.find_by(logingov_uuid: user_uuid) if user_uuid
+        account ||= Account.find_by(edipi: edipi) if edipi
+        # Should we do this instead of expecting a NoMethodError?
+        # raise AccountNotFoundError.new("for user_uuid: #{user_uuid}") unless account&.icn
+        account
+      end
+
+      def bp_readings
+        @bp_readings ||= client.list_resource('observations')
+      end
 
-    def medications(client)
-      @medications ||= client.list_resource('medications')
-      @medications.present? ? FastTrack::HypertensionMedicationRequestData.new(@medications).transform : []
+      def observation_data
+        @observation_data ||=
```

Comment:
I'm not sure why we're memoizing this. I think it just adds complexity. Am I missing a reason?

Reply:
`observation_data` is called twice and as a result, `FastTrack::HypertensionObservationData.new(bp_readings).transform` will be called twice. Since `bp_readings` hasn't changed between the 2 calls, the result from `observation_data` will be the same. Memoizing will avoid reinstantiating `HypertensionObservationData` and re-calling `transform` to get the same result.

Reply:
Ah - I see. I'm neutral here. I think I'd rather it be pulled out into it's own method and only called once. Again, just because I feel like memoizing here is a little extra. All stylistic though so 🤷‍♀️